cognite/client/_api/functions.py

```diff
@@ -716,15 +716,16 @@ def status(self) -> FunctionsStatus:
         return FunctionsStatus.load(res.json())
 
 
-def get_handle_function_node(file_path: Path) -> ast.FunctionDef | None:
-    return next(
-        (
-            item
-            for item in ast.walk(ast.parse(file_path.read_text()))
-            if isinstance(item, ast.FunctionDef) and item.name == "handle"
-        ),
-        None,
-    )
+def get_handle_function_node(file_path: Path) -> ast.FunctionDef | ast.Name | None:
+    tree = ast.parse(file_path.read_text())
+    for node in tree.body:
+        if isinstance(node, ast.FunctionDef) and node.name == "handle":
+            return node
+        if isinstance(node, ast.Assign):
+            for target in node.targets:
+                if isinstance(target, ast.Name) and target.id == "handle":
+                    return target
+    return None
```

Comment:
I wonder (as a follow-up) if this function should not short-circuit on first found and rather raise if > 1 is found. Wdyt @dbrattli ?

Reply:
I think we should not pretend we can statically validate the Python code being deployed. Anything can be wrong (or right) with the function. Checking that it has at least a `handle` is fine with me, but trying to evaluate here if the code will fly or not is over-engineering imo. I.e the existing logic will reject valid code like a callable `handle` since it's not a `FunctionDef`. Why this strict requirement to begin with? If the user deploys a function with two things called `handle`, then I'm good if it works. E.g if a route is also called `handle`, ref: https://github.com/cognitedata/cognite-typed-functions/blob/main/README.md

Reply:
`inspect.getsource` could grab the wrong code, that's the point I'm trying to make! Anyway, I agree on the point re. validation - that can never be bullet-proof.

However, as recent incidents have shown, the FunctionsAPI is subjective to heavy load, and thus keeping errors from triggering a new deployment is for sure valuable.

> Why this strict requirement to begin with?

YAGNI - _until you do_, which seems to be now!

Let me end this reply with a friendly tease regrading over-engineering; Cognite Typed Functions sure looks nice 😉  ( ❤️ )

Reply:
Reverted changes